smartlogic/healthcheck.go

```
@@ -90,36 +100,49 @@ func (h *SmartlogicConcordanceTransformerHandler) concordanceRwNeo4jHealthCheck(
 		clientError := fmt.Sprintf("Error calling writer at %s : %v", urlToCheck, err)
-		log.WithError(err).Error(clientError)
-		return clientError, errors.New("Unable to verify availibility of concordances-rw-neo4j")
+		h.log.WithError(err).Error(clientError)
+		return clientError, errors.New("unable to verify availibility of concordances-rw-neo4j")
 	}
-	defer resp.Body.Close()
-	if resp != nil && resp.StatusCode != http.StatusOK {
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			h.log.WithError(err).Info("Could not close body")
+		}
+	}(resp.Body)
+	if resp.StatusCode != http.StatusOK {
 		clientError := fmt.Sprintf("Writer %v returned status %d", urlToCheck, resp.StatusCode)
-		log.WithError(err).Error(clientError)
-		return clientError, errors.New("Unable to verify availibility of concordances-rw-neo4j")
+		h.log.WithError(err).Error(clientError)
+		return clientError, errors.New("unable to verify availibility of concordances-rw-neo4j")
 	}
 	return "Successfully connected to Concordances Rw Neo4j", nil
 }
 
-func (h *SmartlogicConcordanceTransformerHandler) checkKafkaConnectivity() (string, error) {
+func (h *ConcordanceTransformerHandler) checkKafkaConnectivity() (string, error) {
 	err := h.consumer.ConnectivityCheck()
 	if err != nil {
-		clientError := fmt.Sprint("Error verifying open connection to Kafka")
-		log.WithError(err).Error(clientError)
+		clientError := "error verifying open connection to Kafka"
+		h.log.WithError(err).Error(clientError)
 		return "Error connecting with Kafka", errors.New(clientError)
-	} else {
-		return "Successfully connected to Kafka", nil
 	}
+	return "Successfully connected to Kafka", nil
+}
+
+func (h *ConcordanceTransformerHandler) monitorKafkaConnectivity() (string, error) {
+	if err := h.consumer.MonitorCheck(); err != nil {
+		clientError := "kafka consumer is lagging"
+		h.log.WithError(err).Error(clientError)
+		return "Error communicating with Kafka", errors.New(clientError)
```

This is masking the actual error making it harder for us to investigate. Let's instead return `err` and do also the same for `checkKafkaConnectivity()`.